src/models/decisions/verification.ts

```diff
@@ -13,4 +13,14 @@ export class Verification extends Decision {
   public constructor(confirmed: EnumValue<DecisionOptions>, tag?: Tag) {
     super(confirmed, tag);
   }
+
+  /** returns a new verification instance with the tag property changed */
+  public addTag(tag: Tag): Verification {
```

Comment:
nit: 
```suggestion
  public withTag(tag: Tag): Verification {
```

`with` is commonly used when dealing with records and implies a new record is created